engine/src/main/java/org/terasology/rendering/dag/nodes/DownSamplerNode.java

```diff
@@ -58,28 +57,24 @@ public void initialise() {
     /**
      * Initializes the DownSamplerNode instance. This method is meant to be called once, shortly after instantiation.
      *
-     * @param inputConfig an FBOConfig instance describing the input FBO, to be retrieved from the FBO manager
-     * @param inputManager the FBO manager from which to retrieve the input FBO
-     * @param outputConfig an FBOConfig instance describing the output FBO, to be retrieved from the FBO manager
-     * @param outputManager the FBO manager from which to retrieve the output FBO
+     * @param inputFboConfig an FBOConfig instance describing the input FBO, to be retrieved from the FBO manager
+     * @param inputFboManager the FBO manager from which to retrieve the input FBO
+     * @param outputFboConfig an FBOConfig instance describing the output FBO, to be retrieved from the FBO manager
+     * @param outputFboManager the FBO manager from which to retrieve the output FBO
      * @param aLabel a String to label the instance's entry in output generated by the PerformanceMonitor
      */
-    public void initialise(FBOConfig inputConfig, BaseFBOsManager inputManager,
-                           FBOConfig outputConfig, BaseFBOsManager outputManager,
+    public void initialise(FBOConfig inputFboConfig, BaseFBOsManager inputFboManager,
+                           FBOConfig outputFboConfig, BaseFBOsManager outputFboManager,
                            String aLabel) {
+        this.outputFboManager = outputFboManager;
+        this.outputFboUrn = outputFboConfig.getName();
 
-        this.outputFBOconfig = outputConfig;
-        this.outputFBOmanager = outputManager;
-
-        requiresFBO(inputConfig, inputManager);
-        requiresFBO(outputConfig, outputManager);
-
-        outputFBO = outputManager.get(this.outputFBOconfig.getName());
-        outputManager.subscribe(this);
+        requiresFBO(inputFboConfig, inputFboManager);
+        outputFbo = requiresFBO(outputFboConfig, outputFboManager);
```

Comment:
I was the one who added the update() function to other nodes, most of them didn't even have an update() earlier.
However, the reason I have not used update() in CopyImageToHMDNode and in DownSamplerNode is because it isn't applicable.

Over here, requiresFbo() creates the Fbo, and returns it's FboId. It would be inefficient to ignore this return value, and then ask the fboManager for it's fboId - hence i broke the pattern at times.
Similarly in CopyImageToHMDNode, leftEye and rightEye are generated using requireFbo(), no need to query them again using update().